scripts/demo/sampling.py

```diff
@@ -251,30 +209,34 @@ def apply_refiner(
 
 if __name__ == "__main__":
     st.title("Stable Diffusion")
-    version = st.selectbox("Model Version", list(VERSION2SPECS.keys()), 0)
-    version_dict = VERSION2SPECS[version]
+    version = st.selectbox(
+        "Model Version",
+        [member.value for member in ModelArchitecture],
+        0,
+    )
+    version_enum = ModelArchitecture(version)
+    specs = model_specs[version_enum]
     mode = st.radio("Mode", ("txt2img", "img2img"), 0)
     st.write("__________________________")
 
     set_lowvram_mode(st.checkbox("Low vram mode", True))
 
-    if version.startswith("SDXL-base"):
+    if str(version).startswith("stable-diffusion-xl"):
```

Comment:
I'd probably add a `is_xl` `@property` on `ModelArchitecture` to keep things less stringly typed.